sds_data_manager/lambda_code/SDSCode/path_helper.py

```diff
@@ -0,0 +1,186 @@
+        (
+            self.mission,
+            self.instrument,
+            self.data_level,
+            self.descriptor,
+            self.startdate,
+            self.enddate,
+            self.version,
+            self.extension,
+        ) = self.split_filename
+        # This message is returned to user through API to indicate why filename was not
+        # correct.
+        self.message = None
+
+    def check_date_input(self, input_date: str) -> bool:
+        """Check input date string is in valid format and is correct date
+
+        Parameters
+        ----------
+        input_date : str
+            Date in YYYYMMDD format.
+
+        Returns
+        -------
+        bool
+            Whether date input is valid or not
+        """
+        # Check if the pattern matches 8 digits (YYYYMMDD)
+        if not re.match(r"^\d{8}$", input_date):
+            return False
+
+        # Validate if it's a real date
+        try:
+            datetime.strptime(input_date, "%Y%m%d")
+            return True
+        except ValueError:
+            return False
+
+    def validate_filename(self) -> bool:
+        """Validate filename patterns.
+
+        Returns
+        -------
+        bool
+            Whether filename format is valid or not
+        """
+        pattern = FilenamePatternConfig()
+
+        # Dictionary to map fields to their valid values and error messages
+        validation_checks = {
+            "mission": (self.mission == pattern.mission, "Invalid mission."),
+            "instrument": (
+                self.instrument in pattern.instruments,
+                f"Invalid instrument. Please choose from {pattern.instruments}",
+            ),
+            "data_level": (
+                self.data_level in pattern.data_level,
+                f"Invalid data level. Please choose from {pattern.data_level}",
+            ),
+            "descriptor": (self.descriptor is not None, "Descriptor is required."),
+            "startdate": (
+                self.check_date_input(self.startdate),
+                "Invalid start date format. Please use YYYYMMDD format.",
+            ),
+            "enddate": (
+                self.check_date_input(self.enddate),
+                "Invalid end date format. Please use YYYYMMDD format.",
+            ),
+            "version": (
+                bool(re.match(pattern.version, self.version)),
+                "Invalid version format. Please use vxx-xx format.",
+            ),
+            "extension": (
+                self.extension in pattern.extensions,
+                f"Invalid extension. Please choose from {pattern.extensions}",
+            ),
+        }
+
+        # Iterate through each validation check
+        for _field, (is_valid, error_message) in validation_checks.items():
+            if not is_valid:
+                self.message = error_message
+                return False
+
+        return True
+
+    def create_path_to_upload(self) -> str:
```

Comment:
Rather than just creating the path to the upload, should we just create the entire s3 uri here? We have the filename too, so I think that would save us the headache later of having to append the filename to this path.

Reply:
Will do that.